web/src/AnalyticsNudge.tsx

```diff
@@ -1,33 +1,75 @@
-import React, { PureComponent } from "react"
+import React, { Component } from "react"
 import "./AnalyticsNudge.scss"
-import { ResourceView } from "./types"
 
 type AnalyticsNudgeProps = {
   needsNudge: boolean
 }
 
-class AnalyticsNudge extends PureComponent<AnalyticsNudgeProps> {
-  static analyticsOpt(optIn: boolean) {
+type AnalyticsNudgeState = {
+  requestMade: boolean
+  responseCode: number
+  dismissed: boolean
+}
+
+// constructor for defaults?
+
+class AnalyticsNudge extends Component<
+  AnalyticsNudgeProps,
+  AnalyticsNudgeState
+> {
+  shouldShow(): boolean {
+    return (
+      this.props.needsNudge ||
+      (this.state && this.state.requestMade && !this.state.dismissed)
```

Comment:
Let's set these to sane defaults in a constructor so we don't have to do these null checks.